### DISCUSSION TITLE
Cloudflare images support

### DISCUSSION THREAD

**Discussion by zizther** [spacecatninja/craft-imager-x#240] [General]
With the news that Cloudflare is merging Images and Image Resizing as of November 15th and offering a zero-storage plan for developers who only want to resize and optimise images, which enables you to transform images while keeping the existing storage. Would it be possible for Imager X to support Cloudflare as an image transform with the option to keep original files locally or on S3 for example, or with the option to store images on Cloudflare, through their paid tier.

More information on the Cloudflare merge here
https://blog.cloudflare.com/merging-images-and-image-resizing/

**Comment by rjgux** [comment]
We would also love to be able to use Cloudflare Images with Imager X. Any thoughts on support?

**Comment by plcdnl** [comment]
Any news ?

**Comment by boboldehampsink** [comment]
This would be amazing

**Comment by aelvan** [comment]
I'm working on a transformer for Cloudflare Images this summer. 

For anyone still following this thread; are you planning on using this with assets stored on Cloudflare (Images), or elsewhere? Since I'm not using Cloudflare myself, I'm struggling a bit to decide how I should set up the configuration for typical use-cases.

**Reply by zizther** [reply]
For me, it’s a combination of both. I have existing sites that currently use S3, and I’d like to explore using Cloudflare for asset transforms on those. For new sites, I plan to host and handle asset transformations directly through Cloudflare.

**Comment by aelvan** [comment]
I've created a transformer for Cloudflare Images, see https://github.com/spacecatninja/craft-imager-x-cloudflare-images-transformer (submitted to the plugin store, but not approved yet).

It was a pretty easy transformer to make. The feature set is pretty limited, as is the case with most CDN-based image transform services, but the ones it has mapped relatively well to Imagers features. For instance, it supports crop, fit and letterbox transform modes, and even respects allowUpscale reasonably well. `format=auto` and `gravity=auto` is also pretty awesome and works well. See my notes about this in the `defaultParams` config setting.

Hope you find this useful.

**Reply by zizther** [reply]
This is amazing, thanks for sorting so quickly!
